graphql_compiler/schema/sqlalchemy_schema.py

```diff
@@ -1,17 +1,111 @@
 # Copyright 2019-present Kensho Technologies, LLC.
 from collections import namedtuple
 
+from graphql.type.definition import GraphQLInterfaceType, GraphQLObjectType
+import six
+import sqlalchemy
+
+from . import is_vertex_field_name
+
+
+# Describes the intent to join two tables on equality of a column of theirs.
+# The resulting join expression could be something like:
+# JOIN origin_table.from_column = destination_table.to_column
+#
+# The type of join (inner vs left, etc.) is not specified.
+# The tables are not specified.
+DirectJoinDescriptor = namedtuple('DirectJoinDescriptor', (
+    'from_column',  # The column in the source table we intend to join on.
+    'to_column',    # The column in the destination table we intend to join on.
+))
+
+
+# TODO(bojanserafimov): Define a JunctionJoinDescriptor
+
 
 # Complete schema information sufficient to compile GraphQL queries to SQLAlchemy
 #
-# It describes the tables that correspond to each type, and gives instructions on how
-# to perform joins for each vertex field. The property fields on each type are implicitly
-# mapped to columns with the same name on the corresponding table.
+# It describes the tables that correspond to each type (object type, interface type or union type),
```

Comment:
I don't think union types should be mapped to tables. (I think you only mention objects and interfaces below).